integrated_channels/integrated_channel/tasks.py

```diff
@@ -89,6 +92,36 @@ def _log_batch_task_finish(task_name, channel_code, job_user_id,
         ))
 
 
+def set_enterprise_customer_site_to_request(enterprise_customer):
+    """
+    Set the site of the current request to the site of the given enterprise customer.
+
+    This is necessary since during the transmission of learner data, the request's site is used to determine
+    the domain for LMS API calls. This ensures that the correct site configurations are used when making requests 
+    to the LMS.
+    """
+
+    request = get_current_request()
+    if not request:
+        LOGGER.warning('No current request found in context.')
+        return
```

Comment:
I don't understand this. If request is None, why is this returning? If there is no request, there is no site, therefore you won't be able to get the LMS_INTERNAL_ROOT_URL from the site, correct? @alexjmpb

Reply:
Yes correct, we just use the value for LMS_INTERNAL_ROOT_URL in the django settings.

Reply:
With pearson-core functionality the request object it's always present in the tasks, this would check the case where, for example, pearson-core is not installed.

Reply:
Yes, but my point is that in case request is None for whatever reason, shouldn't this method set the request, set the site, so ThirdPartyAuthApiClient will have the correct site based on the enterprise customer assigned to the integrated channel? @alexjmpb

Reply:
I think that by doing so we are replicating the logic inside pearson-core and if we do so, I think it would be better to create a request object, put the site on it and always set it as the current request.

Reply:
@Squirrel18 The site awareness would depend entirely on pearson-core, but if pearson core is not present and there's no request, then the task will use the URL set in LMS_INTERNAL_ROOT_URL and not the site configuration. Regarding the integration with the command, yes when running the transmission command the pearson core applies the request to the task, but with no site, because there's a request, the function will set the enterprise customer site.